### ISSUE TITLE
problem adding a front port to a device with a rear port..

### ISSUE THREAD

**Issue by ThomasADavis** [netbox-community/ansible_modules#262]
##### ISSUE TYPE

 - Bug Report
###### Ansible:

ansible 2.9.10
  config file = /home/tdavis/gitlab/netbox/ansible.cfg
  configured module search path = ['/home/tdavis/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/tdavis/gitlab/netbox/venv/lib64/python3.6/site-packages/ansible
  executable location = /home/tdavis/gitlab/netbox/venv/bin/ansible
  python version = 3.6.8 (default, Apr  2 2020, 13:34:55) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]

###### Netbox:

netbox-2.8.6

###### Collection:

latest github release

##### SUMMARY

```
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /tmp `"&& mkdir /tmp/ansible-tmp-1594171920.3645124-64740-1987896491334 && echo ansible-tmp-1594171920.3645124-64740-1987896491334="` echo /tmp/ansible-tmp-1594171920.3645124-64740-1987896491334 `" ) && sleep 0'
Using module file /home/tdavis/.ansible/collections/ansible_collections/netbox/netbox/plugins/modules/netbox_front_port.py
<127.0.0.1> PUT /home/tdavis/.ansible/tmp/ansible-local-64664ltlmutsv/tmpmxd8bk2v TO /tmp/ansible-tmp-1594171920.3645124-64740-1987896491334/AnsiballZ_netbox_front_port.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /tmp/ansible-tmp-1594171920.3645124-64740-1987896491334/ /tmp/ansible-tmp-1594171920.3645124-64740-1987896491334/AnsiballZ_netbox_front_port.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/home/tdavis/gitlab/netbox/venv/bin/python3 /tmp/ansible-tmp-1594171920.3645124-64740-1987896491334/AnsiballZ_netbox_front_port.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /tmp/ansible-tmp-1594171920.3645124-64740-1987896491334/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
  File "/tmp/ansible_netbox_front_port_payload_q68ig7sx/ansible_netbox_front_port_payload.zip/ansible_collections/netbox/netbox/plugins/module_utils/netbox_utils.py", line 468, in _nb_endpoint_get
    response = nb_endpoint.get(**query_params)
  File "/home/tdavis/gitlab/netbox/venv/lib64/python3.6/site-packages/pynetbox/core/endpoint.py", line 143, in get
    "get() returned more than one result. "
failed: [localhost] (item={'host': 'pp-e3-r10-u31-c1', 'name': 'LC-2', 'index': 2}) => {
    "ansible_loop_var": "item",
    "changed": false,
    "invocation": {
        "module_args": {
            "data": {
                "description": null,
                "device": "pp-e3-r10-u31-c1",
                "name": "LC-2",
                "rear_port": "MPO",
                "rear_port_position": 2,
                "tags": null,
                "type": "lc"
            },
            "netbox_token": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "netbox_url": "https://netbox.nersc.gov",
            "query_params": null,
            "state": "present",
            "validate_certs": true
        }
    },
    "item": {
        "host": "pp-e3-r10-u31-c1",
        "index": 2,
        "name": "LC-2"
    },
    "msg": "More than one result returned for rear_port"
}
```

##### STEPS TO REPRODUCE

1) create device.
2) add rear port with multiple positions.
3) try to add front port to 1 position.

```yaml
  - name: add front ports for the cassette
    netbox_front_port:
      netbox_url: "{{ netbox_url }}"
      netbox_token: "{{ netbox_token }}"
      data:
        device: "{{ item.host }}"
        name: "{{ item.name }}"
        type: lc
        rear_port: 'MPO'
        rear_port_position: "{{ item.index }}"
      state: present
    loop: "{{ frontports }}"
    connection: localhost

```

##### EXPECTED RESULTS

add front port to rear port with multiple indexes.

##### ACTUAL RESULTS

Failure.

**Comment by FragmentedPacket** [comment]
What if you change `rear_port` to a dictionary?

```yaml
rear_port:
  device: "{{ item.host }}"
  name: "MPO"
```

**Comment by ThomasADavis** [comment]
So, the doc in netbox_front_port is kinda vague about this.  

Using the dict does fix it.  maybe add/change the doc to add the dict for rear_port lookup?  or is this a bug in that isn't using the already supplied device?

```
TASK [add front ports for the MPO cassette] *************************************************************************************************************************************
task path: /home/tdavis/gitlab/netbox/build/test.yml:30
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: tdavis
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /tmp `"&& mkdir /tmp/ansible-tmp-1594268627.6575649-20658-240905824524993 && echo ansible-tmp-1594268627.6575649-20658-240905824524993="` echo /tmp/ansible-tmp-1594268627.6575649-20658-240905824524993 `" ) && sleep 0'
Using module file /home/tdavis/.ansible/collections/ansible_collections/netbox/netbox/plugins/modules/netbox_front_port.py
<127.0.0.1> PUT /home/tdavis/.ansible/tmp/ansible-local-20590qrkko4_y/tmp5bwzo5rm TO /tmp/ansible-tmp-1594268627.6575649-20658-240905824524993/AnsiballZ_netbox_front_port.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /tmp/ansible-tmp-1594268627.6575649-20658-240905824524993/ /tmp/ansible-tmp-1594268627.6575649-20658-240905824524993/AnsiballZ_netbox_front_port.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/home/tdavis/gitlab/netbox/venv/bin/python3 /tmp/ansible-tmp-1594268627.6575649-20658-240905824524993/AnsiballZ_netbox_front_port.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /tmp/ansible-tmp-1594268627.6575649-20658-240905824524993/ > /dev/null 2>&1 && sleep 0'
changed: [localhost] => {
    "changed": true,
    "diff": {
        "after": {
            "state": "present"
        },
        "before": {
            "state": "absent"
        }
    },
    "front_port": {
        "cable": null,
        "description": "",
        "device": 325,
        "id": 331,
        "name": "LC-1",
        "rear_port": 13,
        "rear_port_position": 1,
        "tags": [],
        "type": "lc"
    },
    "invocation": {
        "module_args": {
            "data": {
                "description": null,
                "device": "pp-e3-r10-u31-c1",
                "name": "LC-1",
                "rear_port": {
                    "device": "pp-e3-r10-u31-c1",
                    "name": "MPO"
                },
                "rear_port_position": 1,
                "tags": null,
                "type": "lc"
            },
            "netbox_token": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "netbox_url": "https://netbox.nersc.gov",
            "query_params": null,
            "state": "present",
            "validate_certs": true
        }
    },
    "msg": "front_port LC-1 created"
}
META: ran handlers
META: ran handlers

PLAY RECAP **********************************************************************************************************************************************************************
localhost                  : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=1   
```

**Comment by toerb** [comment]
It seems I have missed it when implementing the module.
From my point of view the already provided devices should be used, since a **front port** can only be connected to the **rear port of the same device**.

I will have a look into it and extend the tests around that a little bit.

**Comment by FragmentedPacket** [comment]
I think I do the same for `lag` so we can do the same for this as well.

**Comment by FragmentedPacket** [comment]
The fix was merged in and should be released with 0.2.4
